### PR TITLE
Send post every week

### PR DESCRIPTION
**Pull request by manland**

### CODE REVIEW

File: plugin.json [hunk 1]
```diff
@@ -3,6 +3,7 @@
     "name": "Analytics matter",
     "description": "This plugin give analytics of your mattermost instance to your users.",
     "version": "0.0.1",
+    "min_server_version": "5.6.0",
```
Comment (hanzei) [inline plugin.json:6]: :+1:

File: plugin.json [hunk 2]
```diff
@@ -11,8 +12,32 @@
...
-        "header": "",
+        "header": "Analytics bot configuration",
         "footer": "",
-        "settings": []
+        "settings": [
+            {
+                "key": "Username",
+                "display_name": "User",
+                "type": "username",
+                "help_text": "Select the username of the user that the plugin will post with. This can be any user, the name and icon will be overridden when posting."
```
Comment (hanzei) [inline plugin.json:22]: Maybe out of scope here, but what do you think of using a bot account for this? You can find the spec [here](https://docs.google.com/document/d/139dNmjHz0Tflcgby85xaJtxYwapPctZHiZpZT6aS0DA).

File: server/activate.go [hunk 3]
```diff
@@ -93,7 +101,11 @@ func (p *Plugin) handleLine(w http.ResponseWriter, r *http.Request) {
 		} else {
 			yvalue := make([]float64, 0)
 			for _, value := range values {
-				v, _ := strconv.ParseFloat(value, 64)
+				v, err := strconv.ParseFloat(value, 64)
+				if err != nil {
+					p.API.LogError("can't parse value", "value", value, "url", r.URL.String())
+					v = 0
```
Comment (hanzei) [inline server/activate.go:112]: Don't you want to `return` here?

File: server/configuration.go [hunk 4]
```diff
@@ -79,5 +106,35 @@ func (p *Plugin) OnConfigurationChange() error {
...
+		return err
+	}
+
+	user, err := p.API.GetUserByUsername(configuration.Username)
+	if err != nil {
+		return fmt.Errorf("Unable to find user with configured username: %v", configuration.Username)
+	}
+	p.BotUserID = user.Id
+
+	channelsID := make([]string, 0)
```
Comment (hanzei) [inline server/configuration.go:129]: 1/5 that L119-137 should go into there own method.

File: server/post.go [hunk 5]
```diff
@@ -0,0 +1,139 @@
...
+
+import (
+	"fmt"
+	"net/url"
+
+	"github.com/mattermost/mattermost-server/model"
+	"github.com/robfig/cron"
+)
+
+func (p *Plugin) startCron() error {
```
Comment (hanzei) [inline server/post.go]: Maybe move `startCron()` into `plugin.go` and leave L13-17 in this file as a own method

File: server/post.go [hunk 6]
```diff
@@ -0,0 +1,139 @@
...
+	m, err := p.buildAnalyticMsg()
+	if err != nil {
+		return err
+	}
+	for _, channelID := range p.ChannelsID {
+		post := &model.Post{
+			UserId:    p.BotUserID,
+			ChannelId: channelID,
+			Message:   m,
+			Type:      "custom_git_welcome",
```
Comment (hanzei) [inline server/post.go]: Why do you need this?

File: server/post.go [hunk 7]
```diff
@@ -0,0 +1,139 @@
...
+}
+
+func (p *Plugin) buildAnalyticMsg() (string, *model.AppError) {
+	data, err := p.prepareData()
+	if err != nil {
+		return "", err
+	}
+
+	p.currentAnalytic.RLock()
+	m := fmt.Sprintf("## Analytics since %s at %s\n", p.currentAnalytic.Start.Format("2 January"), p.currentAnalytic.Start.Format("15:04"))
```
Comment (hanzei) [inline server/post.go:17]: Presumably out of scope here, but what do you think about using https://docs.mattermost.com/developer/message-attachments.html for formatting the message?

File: plugin.json (hunk 2 above)
Comment (manland) [reply plugin.json:22]: O :heart_eyes: exactly what i need ! https://github.com/manland/mattermost-plugin-analytics/issues/20

File: server/activate.go (hunk 3 above)
Comment (manland) [reply server/activate.go:112]: No i want to draw a chart even if a value is not good. In the philosophie of `draw something even if its not the truth`

File: server/configuration.go (hunk 4 above)
Comment (manland) [reply server/configuration.go:129]: :heavy_check_mark:

File: server/post.go (hunk 5 above)
Comment (manland) [reply server/post.go]: I have done a cron file containing all cron stuff, it will come when I push ;)

File: server/post.go (hunk 6 above)
Comment (manland) [reply server/post.go]: Good catch, it's a bad copy/paste from an example I have found... :sweat_smile:

File: server/post.go (hunk 7 above)
Comment (manland) [reply server/post.go:17]: Thanks, i have not found this doc, I will use tabulation representation! https://github.com/manland/mattermost-plugin-analytics/issues/22

**Pull request merged**